src/index.js

```diff
@@ -0,0 +1,565 @@
+      maxTimeout = c.minute
+    }
+
+    this._log('findPeer %s', id.toB58String())
+
+    this.findPeerLocal(id, (err, pi) => {
+      if (err) {
+        return callback(err)
+      }
+      // already got it
+      if (pi != null) {
+        this._log('found local')
+        return callback(null, pi)
+      }
+
+      waterfall([
+        (cb) => utils.convertPeerId(id, cb),
+        (key, cb) => {
+          const peers = this.routingTable.closestPeers(key, c.ALPHA)
+          if (peers.length === 0) {
+            return cb(new errors.LookupFailureError())
+          }
+
+          // sanity check
+          const match = peers.find((p) => p.isEqual(id))
+          if (match && this.peerBook.has(id)) {
+            this._log('found in peerbook')
+            return cb(null, this.peerBook.get(id))
+          }
+
+          // query the network
+          const query = new Query(this, id.id, (peer, cb) => {
+            waterfall([
+              (cb) => this._findPeerSingle(peer, id, cb),
+              (msg, cb) => {
+                const match = msg.closerPeers.find((p) => p.id.isEqual(id))
+
+                // found it
+                if (match) {
+                  return cb(null, {
+                    peer: match,
+                    success: true
+                  })
+                }
+
+                cb(null, {
+                  closerPeers: msg.closerPeers
+                })
+              }
+            ], cb)
+          })
+
+          timeout((cb) => {
+            query.run(peers, cb)
+          }, maxTimeout)(cb)
+        },
+        (result, cb) => {
+          this._log('findPeer %s: %s', id.toB58String(), result.success)
+          if (!result.peer) {
+            return cb(new errors.NotFoundError())
+          }
+          cb(null, result.peer)
+        }
+      ], callback)
+    })
+  }
+
+  /**
+   * Look if we are connected to a peer with the given id.
+   * Returns the `PeerInfo` for it, if found, otherwise `undefined`.
+   *
+   * @param {PeerId} peer
+   * @param {function(Error, PeerInfo)} callback
+   * @returns {void}
+   */
+  findPeerLocal (peer, callback) {
+    this._log('findPeerLocal %s', peer.toB58String())
+    this.routingTable.find(peer, (err, p) => {
+      if (err) {
+        return callback(err)
+      }
+      if (!p || !this.peerBook.has(p)) {
+        return callback()
+      }
+      callback(null, this.peerBook.get(p))
+    })
+  }
+
+  /**
+   * Start the bootstrap process. This means running a number of queries every interval requesting random data.
+   * This is done to keep the dht healthy over time.
+   *
+   * @param {number} [queries=1] - how many queries to run per period
+   * @param {number} [period=300000] - how often to run the the bootstrap process, in milliseconds (5min)
+   * @param {number} [maxTimeout=10000] - how long to wait for the the bootstrap query to run, in milliseconds (10s)
+   * @returns {void}
+   */
+  bootstrapStart (queries, period, maxTimeout) {
```

Comment:
This is, in fact, a discovery service that DHT provides and so as any discovery service it needs to be passed into the discovery mechanisms